### PR TITLE
feat(list): added some animation on list

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Enhancement

#### What is expected?

As List was already have an animation I tough it would be cool to have them all as we already did for RichSelect. I will use animation for instance when adding new volume and I think there might be others use case where animation will make console look better.

## Relevant logs and/or screenshots

https://user-images.githubusercontent.com/15812968/157975916-d52e35be-4164-4467-8a98-daaac092f104.mov

https://user-images.githubusercontent.com/15812968/157975930-f793ccdb-adef-4a94-9a77-384832e29d42.mov

### CODE REVIEW

File: src/components/List/__stories__/index.stories.tsx [hunk 1]
```diff
@@ -518,3 +521,80 @@ PaginationLoadingPageCount.decorators = [
     )
   },
 ]
+
+export const Animated = Template.bind({})
```
Comment (adriengibrat) [inline src/components/List/__stories__/index.stories.tsx:525]: Excellent demo ;)
An UX review should be as important as the code review for this!

File: src/components/List/variantProduct.tsx [hunk 2]
```diff
@@ -305,13 +299,15 @@ export const Row: FunctionComponent<ListRowProps> = ({
   id,
   children,
   animated,
-  edition,
-  isEditable,
-  isHoverable,
-  locked,
+  animation = 'fadeIn',
```
Comment (adriengibrat) [inline src/components/List/variantProduct.tsx:302]: should use type `keyof animations`

File: src/components/List/__stories__/index.stories.tsx (hunk 1 above)
Comment (matthprost) [reply src/components/List/__stories__/index.stories.tsx:525]: @adriengibrat Probably animations will be more complete once design system process will start and we will have more from design team :)

File: src/components/List/variantTable.tsx [hunk 3]
```diff
@@ -73,13 +64,22 @@ const StyledRow = styled(Box, {
...
+  animationDuration?: number
+  disabled?: boolean
+  highlighted?: boolean
+}>`
+  ${({ animated, animationDuration, animation }) =>
     animated
       ? css`
-          animation: ${fadeInAnimation} 1s linear;
+          animation: ${(
+              animations as Record<string, ReturnType<typeof keyframes>>
```
Comment (chambo-e) [inline src/components/List/variantTable.tsx:78]: why ?

File: src/components/List/variantTable.tsx [hunk 4]
```diff
@@ -73,13 +64,22 @@ const StyledRow = styled(Box, {
...
+            )[animation as string]}
+            ${animationDuration as number}ms linear;
```
Comment (chambo-e) [inline src/components/List/variantTable.tsx:80]: why ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

**Review by chambo-e — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**